backend/voting/models/ranking.py

```diff
@@ -0,0 +1,171 @@
+from submissions.models import Submission
+
+
+class RankRequest(models.Model):
+
+    conference = models.ForeignKey(
+        "conferences.Conference", on_delete=models.CASCADE, verbose_name=_("conference")
+    )
+
+    created = AutoCreatedField(_("created"))
+
+    def __str__(self):
+        return (
+            f"{self.conference.name} <{self.conference.code}>, "
+            f"created on {self.created}"
+        )
+
+    def save(self, *args, **kwargs):
+        super(RankRequest, self).save(*args, **kwargs)
+
+        ranked_submissions = self.build_ranking(self.conference)
+
+        self.save_rank_submissions(ranked_submissions)
+
+    def build_ranking(self, conference):
+        """Builds the ranking
+
+        P. S. maybe we should make some *Strategy* in order to have all
+        available and run different strategies...
+
+        :return: list of dictionary ordered by score descending
+        [{
+            "submission_id": submission.id,
+            "submission__topic_id": submission.topic_id,
+            "score": score
+        },
+        ...
+        ]
+        """
+
+        return RankRequest.users_most_voted_based(conference)
+        # return RankRequest.simple_sum(conference)
+
+    @staticmethod
+    def users_most_voted_based(conference):
+        """Builds the ranking based on the votes each user has gived
+        This algorithm rewards users who have given more votes. If a user
+        votes many submissions, it means that he cares about his choices so
+        he must be rewarded by weighing his votes more.
+
+        P. S. maybe we should make some *Strategy* in order to have all
+        available and run different strategies...
+
+        """
+
+        from voting.models import Vote
+
+        submissions = Submission.objects.filter(conference=conference)
+        votes = Vote.objects.filter(submission__conference=conference)
+
+        users_weight = RankRequest.get_users_weights(conference)
+
+        ranking = []
+        for submission in submissions:
+            submission_votes = votes.filter(submission=submission)
+
+            vote_info = {}
+            for vote in submission_votes:
+                vote_info[vote.id] = {
+                    "normalised_vote": vote.value * users_weight[vote.user.id],
+                    "scale_factor": users_weight[vote.user.id],
+                }
+            score = sum([v["normalised_vote"] for v in vote_info.values()]) / sum(
+                [v["scale_factor"] for v in vote_info.values()]
+            )
+            rank = {
+                "submission_id": submission.id,
+                "submission__topic_id": submission.topic_id,
+                "score": score,
+            }
+            ranking.append(rank)
+        return sorted(ranking, key=lambda k: k["score"], reverse=True)
+
+    @staticmethod
+    def get_users_weights(conference):
+        from users.models import User
+
+        users = User.objects.all()
```

Comment:
To reduce the hit to the Database and retain only interesting users I would rather use the `submission` query set and get all the unique users from that.
I don't know if this **actually** works - should be tested - but something like:

```
user_ids = submissions.distinct('user').values_list('user', flat=True)
voters = User.objects.filter(id__in=user_ids)
```